### PR TITLE
feat: support classic ingest keys

### PR DESCRIPTION
**Pull request by brookesargent**
## Which problem is this PR solving?

We've released Ingest Keys, but need to update the key detection logic to allow Ingest Keys to be used to send data to Classic environments.

## Short description of the changes
- updates the Classic API Key detection logic to understand the shape of a Classic Ingest Key
- updates tests with a more comprehensive list of cases
- bonus: updated the circleCI image for smoke tests to `ubuntu-2204:2024.01.1` since the one we were using [is deprecated](https://discuss.circleci.com/t/linux-image-deprecations-and-eol-for-2024/50177)

## How to verify that this has the expected result

It should be possible to use [an ingest key](https://docs.honeycomb.io/working-with-your-data/settings/api-keys/#ingest-keys-1) pointed to a classic environment after this change. Please note that the ability to create ingest keys in classic environments isn't yet publicly available (since we need to update this library and others to understand classic ingest keys).

### CODE REVIEW

**Review by jharley — APPROVED** [review]

**Comment by cartermp** [comment]
wait wtf where did my suggestion go 💀

File: src/Honeycomb.OpenTelemetry/HoneycombOptions.cs [hunk 1]
```diff
@@ -48,13 +51,26 @@ public class HoneycombOptions
...
+            else if (apikey.Length == 32)
+            {
+              return Regex.Match(apikey, ClassicKeyRegex).Success;
+            }
+            else if (apikey.Length == 64)
+            {
+              return Regex.Match(apikey, IngestClassicKeyRegex).Success;
+            }
+            return false;
+        }
```
Comment (cartermp) [inline src/Honeycomb.OpenTelemetry/HoneycombOptions.cs:73]:
```suggestion
        internal static bool IsClassicKey(string apikey)
        {
            if (String.IsNullOrEmpty(apikey))
            {
                return false;
            }
            else if (apikey.Length == 32)
            {
                return Regex.Match(apikey, ClassicKeyRegex).Success;
            }
            else if (apikey.Length == 64)
            {
                return Regex.Match(apikey, IngestClassicKeyRegex).Success;
            }
            return false;
        }
```

**Review by cartermp — COMMENTED** [review]
code style change suggestion again?

**Review by MikeGoldsmith — APPROVED** [review]

**Review by MikeGoldsmith — APPROVED** [review]
Looks good, thanks @brookesargent.

I merged the suggested formatting from @cartermp plus updated the smoke test image that needed updating.

**Pull request merged**